firmware/applications/SignBuddy/comms.c

```diff
@@ -0,0 +1,164 @@
+#include "comms.h"
+
+#include "ble_uart.h"
+#include "board.h"
+#include "common.h"
+#include "crc.h"
+#include "fifo.h"
+#include "logger.h"
+#include "pb_encode.h"
+#include "sensors.h"
+#include "SignBuddy.pb.h"
+
+#include <stdlib.h>
+
+#define COMMS_TX_BUFFER_SIZE    512
+
+#define SYNC_BEGIN              0x16
+#define SYNC_END                0x27
+
+#define MID_SAMPLE              0x01
```

Comment:
https://developers.google.com/protocol-buffers/docs/proto3#enum

we should just make a message type enum in protobuf and include that in the packet

Reply:
I don't see how this would work because the order of the bytes isn't guaranteed, we wouldn't know which byte is the mid when reconstructing on the android app

Reply:
You just decode the protobuf message and then check which type it is

You can also do it like this: 

https://stackoverflow.com/questions/30564404/how-to-determine-message-type-in-protobuf-so-that-i-can-use-that-type-parsefrom/30565011

Reply:
Actually nevermind

Reply:
What do you mean by that?

Reply:
I think the method I linked above might not work for us very well

but i think we should define a message id enum in the proto file and then use that as the message id. that way at the very least we share the message id types between android and chip

Reply:
I'll make that change on a different branch